e2e/fuzz/runner.go

```diff
@@ -0,0 +1,115 @@
+package fuzz
+
+import (
+	"log"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/0xPolygon/pbft-consensus"
+	"github.com/0xPolygon/pbft-consensus/e2e"
+)
+
+var (
+	revertThreshold        = 30
+	maxSleepBetweenActions = 10
+)
+
+type Runner struct {
+	wg               sync.WaitGroup
+	cluster          *e2e.Cluster
+	availableActions []e2e.Action
+	// TODO: Add runningActions slice, which will be randomly & dynamically generated and will contain sequence of actions being executed?
+}
+
+func NewRunner(initialNodesCount uint) *Runner {
+	return &Runner{
+		availableActions: getAvailableActions(),
+		cluster:          e2e.NewPBFTCluster(nil, "fuzz_cluster", "NODE", int(initialNodesCount)),
+		wg:               sync.WaitGroup{},
+	}
+}
+
+func (r *Runner) Run(d time.Duration) error {
+	r.cluster.Start()
+	// TODO: we need to stop the cluster, what about tracer inside cluster stop?
+	// defer r.cluster.Stop()
+	done := time.After(d)
+	actionsApplied := make([]int, 0)
+	r.wg.Add(1)
+	go func() {
+		defer r.wg.Done()
+		for {
```

Comment:
```
var reverts []func()

for {
    select {
        case <-time.After(5 * time.Second):
             action := &NodeDropAction{} // use an array once more actions are available
             if action.CanApply(c) {
                  revertFn := action.Apply(c)
                  reverts = append(reverts, revertFn)
             }
        case <-time.After(3 * time.Second):
             if len(reverts) == 0 {
                  continue
             }
             if // 20% chance {
                 var revertFn func()
                 revertFn, reverts = reverts[0], reverts[1:] // pop a revert
                 revertFn()
             }                 
    }
}
```

Reply:
Should we randomize intervals for apply, revert (and possibly validating a cluster) as well or we shouldn't go that way? :slightly_smiling_face: 
However think that there might be some limitations (e.g. validation interval shouldn't be smaller than at least apply interval if not apply+revert interval etc.).